sentry_streams/sentry_streams/pipeline/pipeline.py

```diff
@@ -136,6 +139,27 @@ def add(self, other: Pipeline) -> None:
         for source, dests in other.outgoing_edges.items():
             self.outgoing_edges[source] = dests
 
+    def start(self, step: Source) -> Pipeline:
+        assert not self.__previous_step, "Cannot start a pipeline with multiple sources"
+        step.register(self)
+        self.__previous_step = step
+        return self
+
+    def apply(self, step: Step) -> Pipeline:
+        assert self.__previous_step, "Cannot apply a step without starting first"
+        step.register(self, self.__previous_step)
+        self.__previous_step = step
+        return self
+
+    def sink(self, step: Sink) -> None:
+        assert self.__previous_step, "Cannot create a sink a step without starting first"
+        step.register(self, self.__previous_step)
+        self.__previous_step = None
+
```

Comment:
I think we are missing `route` and `broadcast`. Is that intentional ?

Also in `ExtensibelChain` there is the bulk of the docs we have today. 
- https://github.com/getsentry/streams/blob/main/sentry_streams/sentry_streams/pipeline/chain.py#L265-L292
- https://github.com/getsentry/streams/blob/main/sentry_streams/sentry_streams/pipeline/chain.py#L302-L312

Could you please bring it over? Those docstrings are published in the library docs.